src/libstd/ffi/c_str.rs

```diff
@@ -134,10 +135,9 @@ pub struct CString {
 /// in each pair are borrowed references; the latter are owned
 /// strings.
 ///
-/// Note that this structure is **not** `repr(C)` and is not recommended to be
-/// placed in the signatures of FFI functions. Instead, safe wrappers of FFI
-/// functions may leverage the unsafe [`from_ptr`] constructor to provide a safe
-/// interface to other consumers.
+/// Note that this structure is `repr(C)` and is can be
+/// placed in the signatures of FFI functions. A `&CStr` is equivalent to `*const c_char`,
+/// but offers a richer API.
```

Comment:
Could you add a note in this comment regarding the use of `Option<&CStr>` in FFI to handle potentially null pointers?